src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,119 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        table = new Node[DEFAULT_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (key == null) {
+            putForNullKey(value);
+            return;
+        }
+        int hash = hash(key.hashCode());
+        int index = indexFor(hash, table.length);
+        for (Node<K, V> node = table[index]; node != null; node = node.next) {
+            if (node.hash == hash && (node.key == key || key.equals(node.key))) {
+                node.value = value;
+                return;
+            }
+        }
+        addNode(hash, key, value, index);
     }
 
     @Override
     public V getValue(K key) {
+        if (key == null) {
+            return getForNullKey();
+        }
+        int hash = hash(key.hashCode());
+        int index = indexFor(hash, table.length);
+        for (Node<K, V> node = table[index]; node != null; node = node.next) {
+            if (node.hash == hash && (node.key == key || key.equals(node.key))) {
+                return node.value;
+            }
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void putForNullKey(V value) {
+        for (Node<K, V> node = table[0]; node != null; node = node.next) {
+            if (node.key == null) {
+                node.value = value;
+                return;
+            }
+        }
+        addNode(0, null, value, 0);
+    }
+
+    private V getForNullKey() {
+        for (Node<K, V> node = table[0]; node != null; node = node.next) {
+            if (node.key == null) {
+                return node.value;
+            }
+        }
+        return null;
+    }
```

Comment:
i don't think we should handle null keys separately if we use correct comparison condition like in Objects.equals

Reply:
without this the tests will not pass